jaeger_client/span_context.py

```diff
@@ -49,20 +57,20 @@ def with_baggage_item(self, key, value):
         )
 
     @property
-    def has_trace(self):
-        return self.trace_id and self.span_id and self.flags is not None
+    def has_trace(self) -> bool:
+        return self.flags is not None and self.trace_id and self.span_id  # type: ignore
```

Comment:
I have still no clue why some boolean expressions are not resolved as bools. Let's ignore this for now

Reply:
I don't think they are Boolean: 'and x' in a short circuit situation returns the actual x

Reply:
Of course, thanks 😄